syntax = "proto3";

service ReviewService {
  rpc AddReview (ReviewRequest) returns (Review) {}
  rpc GetReviews (GetReviewsRequest) returns (ReviewsList) {}
}

message ReviewRequest {
  int32 user_id = 1;
  int32 product_id = 2;
  float rating = 3;
  string description = 4;
}


message GetReviewsRequest {
  int32 product_id = 1;
}

message ReviewsList {
  repeated Review reviews = 1;
}

message Review {
  int32 id = 1;
  int32 user_id = 2;
  int32 product_id = 3;
  float rating = 4;
  string description = 5;
  string date_added = 6; 
}